presentation/src/main/java/com/nexters/boolti/presentation/service/BtFirebaseMessagingService.kt

```diff
@@ -0,0 +1,56 @@
+package com.nexters.boolti.presentation.service
+
+import android.Manifest
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.nexters.boolti.domain.repository.AuthRepository
+import com.nexters.boolti.presentation.R
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class BtFirebaseMessagingService : FirebaseMessagingService() {
+    private val scope = CoroutineScope(Dispatchers.IO + SupervisorJob())
+
+    @Inject
+    lateinit var authRepository: AuthRepository
+
+    override fun onNewToken(token: String) {
+        scope.launch {
+            authRepository.sendFcmToken()
+        }
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        if (ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.POST_NOTIFICATIONS
+            ) != PackageManager.PERMISSION_GRANTED
+        ) return
```

Comment:
```suggestion
        if (!checkGranted(Manifest.permission.POST_NOTIFICATIONS)) return
```

확장함수 만들어둔거로 활용해도 좋을듯~